crates/scarb-api/src/artifacts/representation.rs

```diff
@@ -0,0 +1,78 @@
+use crate::artifacts::deserialized::{artifacts_for_package, StarknetArtifacts};
+use anyhow::anyhow;
+use camino::{Utf8Path, Utf8PathBuf};
+
+pub(super) struct StarknetArtifactsRepresentation {
```

Comment:
This struct could be a function, you always use `try_new` and `artifacts` in a sequence anyways. This struct is dropped in the same function it is created in

Reply:
Doesn't hurt us and I prefer to use structs to represent intermediate data than passing random tuples.